packages/cosmic-swingset/test/make.test.ts

```diff
@@ -111,119 +104,121 @@ test.serial('integration test: rosetta CI', async t => {
   t.is(code, 0, 'make scenario2-run-rosetta-ci is successful');
 });
 
-/** @type {import('ava').Macro<[title: string, verifier?: any], TestContext>} */
-const walletProvisioning = test.macro({
-  title(_, title, _verifier) {
-    return title;
-  },
-  async exec(t, _title, verifier) {
-    const retryCountMax = 5;
-    // Resume the chain...
-    const { scenario2, walletTool } = t.context;
-    const { agd, query, waitForBlock } = walletTool;
-
-    // Run the chain until error or this test exits.
-    const chain = scenario2.spawnMake(['scenario2-run-chain'], {
-      stdio: fdList,
-    });
-    t.teardown(async () => {
-      chain.kill();
-      await Promise.allSettled([chain.exit]);
-    });
-
-    await waitForBlock('chain bootstrap', 1);
-
-    // Create a new account under the fresh key's name.  Effectively
-    // anonymous, but still accessible if we later need it.
-    const { address: freshAddr } = await agd([
-      'keys',
-      'add',
-      'doNotStore',
-      '--dry-run',
-      '--output=json',
-    ]);
-
-    const checkWalletExists = async address => {
-      try {
-        const { value } = await query([
-          '--output=json',
-          'vstorage',
-          'path',
-          `published.wallet.${address}`,
-        ]);
-        t.log(
-          `query vstorage path published.wallet.${address} exits successfully`,
-        );
-        return !!value;
-      } catch (e) {
-        t.log(e);
-        return false;
-      }
-    };
+interface Verifier {
+  start: (t: ExecutionContext<TestContext>) => Promise<void>;
+  stop: (t: ExecutionContext<TestContext>) => Promise<void>;
+}
 
-    t.false(
-      await checkWalletExists(freshAddr),
-      `${freshAddr} wallet doesn't exist yet`,
-    );
+const walletProvisioning: Macro<
+  [title: string, verifier?: Verifier],
+  TestContext
+> = test.macro(async (t, _title, verifier) => {
```

Comment:
We can simplify further, as per [AVA macro docs](https://github.com/avajs/ava/blob/main/docs/01-writing-tests.md#reusing-test-logic-through-macros):
```suggestion
const walletProvisioning: Macro<
  [verifier?: Verifier],
  TestContext
> = test.macro(async (t, verifier) => {
```